sdk/program/src/bignum.rs

```diff
@@ -0,0 +1,851 @@
+//! Solana BigNumber
+//!
+//! Rust-based BigNumber provides safe math for arbitrarily large numbers.
+//! Primarily focues on BPF program usage, non-BPF is also supported by
+//! backing from the OpenSSL implementation
+use std::fmt;
+
+#[repr(C)]
+#[derive(Debug)]
+/// BPF structure for sending/receiving BigNumber data to/from syscalls
+pub struct FfiBigNumber {
+    pub data_len: u64,     // Length of bytes of data file
+    pub is_negative: bool, // False (0) or True (1) if data is positive or negative
+    pub data: u64, // Pointer to the variable length big endian array represneting the vallue of BigNumber
+}
+
+#[repr(C)]
+#[derive(Eq, Ord, PartialOrd, AbiExample)]
```

Comment:
Add `Debug` and `Display` here

Reply:
@jackcmay Debug works fine, as does Eq and PartialEq but I get an error on the `Display` from rust documentation:
```
Display is similar to Debug, but Display is for user-facing output, and so cannot be derived.
```